plux/build/setuptools.py

```diff
@@ -39,20 +41,31 @@ def _ensure_dist_info(self, *args, **kwargs):
 class plugins(InfoCommon, setuptools.Command):
     description = "Discover plux plugins and store them in .egg_info"
 
-    user_options = [
-        # TODO
+    user_options: t.ClassVar[t.List[t.Tuple[str, str, str]]] = [
+        ('exclude=', 'e', "exclude those files when discovering plugins"),
+        # TODO: add more
     ]
 
     egg_info: str
 
     def initialize_options(self) -> None:
         self.plux_json_path = None
+        self.exclude = None
 
     def finalize_options(self) -> None:
         self.plux_json_path = get_plux_json_path(self.distribution)
+        self.ensure_string_list('exclude')
+        if self.exclude is None:
+            self.exclude = []
+
+        project_config = read_configuration(self.distribution)
+        file_exclude = project_config.get("exclude")
+        if file_exclude:
+            self.exclude = set(self.exclude) | set(file_exclude)
+        self.exclude = [_path_to_module(item) for item in self.exclude]
 
     def run(self) -> None:
-        plugin_finder = PluginFromPackageFinder(DistributionPackageFinder(self.distribution))
+        plugin_finder = PluginFromPackageFinder(DistributionPackageFinder(self.distribution, exclude=self.exclude))
```

Comment:
I'm actually not too sure if we should filter at the `Package` level or the `Module` level. Module level might be a bit more flexible, right now we have to avoid the full package and specifying a module name won't work. Will update 😅

Reply:
I haven't thought very long about, but I think either can work. This looks good!